src/lib/pathPlanner.ts

```diff
@@ -278,12 +278,12 @@ export async function computeSuggestedSkills(
 	for (let i = 1; i < learningUnits.length; i++) {
 		const previousUnit = learningUnits[i - 1];
 		const currentUnit = learningUnits[i];
-		const missingSkills = previousUnit.teachingGoals
+		const missingSkills = previousUnit.getTeachingGoals()
 			.map(goal => goal.id)
 			// Do not copy hard constraints also to soft constraints
-			.filter(goalId => !currentUnit.requiredSkills.map(skill => skill.id).includes(goalId))
+			.filter(goalId => !currentUnit.getRequiredSkills().map(skill => skill.id).includes(goalId))
 			// Do not copy currently taught skills to avoid cycles
-			.filter(goalId => !currentUnit.teachingGoals.map(skill => skill.id).includes(goalId));
+			.filter(goalId => !currentUnit.getTeachingGoals().map(skill => skill.id).includes(goalId));
```

Comment:
To improve readability and maintainability, I recommend introducing smaller helper variables or functions in this section. This approach not only makes the code more self-explanatory but also leverages a functional programming style that is both elegant and potentially more efficient. Here's an example to illustrate this:

```ts
const isNoTeachingGoal= goalId => !currentUnit.getTeachingGoals().some(skill => skill.id === goalId);
const isNoHardConstraint= goalId => !currentUnit.getRequiredSkills().some(skill => skill.id === goalId);

const missingSkills = previousUnit.getTeachingGoals()
  .map(goal => ({ id: goal.id }))
  .filter(isNoTeachingGoal)
  .filter(isNoHardConstraint);
```

With this refactoring, you achieve several improvements:

  -  Enhanced Code Self-documentation: By assigning clear and descriptive names to variables, the code becomes self-explanatory, eliminating the need for additional comments. May think about the suggested variable names (`isNoHardConstrait`). It can be hard to guess what you're trying to do from the outside. Other suggestion would be `doesNotMeetTeachingGoal` and `lacksRequiredSkill`
  -  Readability: The use of well-named helper functions makes the code flow easier to understand at a glance, adhering to a readable functional programming approach.
  -  Efficiency: Utilizing `Array.prototype.some` allows for short-circuiting logic, which can be more performance-efficient, especially with larger arrays.

Note: remember to use teaching goals as property like `currentUnit.teachingGoals` in the future as mentioned in my other comments.